pkg/lookups/lookups.go

```diff
@@ -150,3 +152,69 @@ func GetAllLookupFiles(ctx *fasthttp.RequestCtx) {
 		return
 	}
 }
+
+func GetLookupFile(ctx *fasthttp.RequestCtx) {
+	lookupFilename := utils.ExtractParamAsString(ctx.UserValue("lookupFilename"))
+
+	lookupsDir := config.GetLookupPath()
+	filePath := filepath.Join(lookupsDir, lookupFilename)
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			ctx.Error("File not found", fasthttp.StatusNotFound)
+		} else {
+			utils.SendInternalError(ctx, "Error while opening the file", "", err)
+		}
+		return
+	}
+	defer file.Close()
+
+	// Set the Content-Type header to indicate it's a CSV file
+	ctx.Response.Header.Set("Content-Type", "text/csv")
+
+	// Create a buffered reader
+	reader := bufio.NewReader(file)
+
+	// Buffer to hold data chunks
+	buf := make([]byte, 8192) // 8KB buffer
+
+	// Stream the file data to the response in chunks
+	for {
+		n, err := reader.Read(buf)
+		if err != nil {
+			if err.Error() != "EOF" {
+				utils.SendInternalError(ctx, "Error while reading the file", "", err)
+				return
+			}
+			break
+		}
+		if _, err := ctx.Write(buf[:n]); err != nil {
+			utils.SendInternalError(ctx, "Error while sending the data", "", err)
+			return
+		}
+	}
```

Comment:
Instead of reading into a buffer and then writing that buffer to `ctx`, I'm pretty sure you can simply use `io.Copy`, which does buffering internally. Also it looks like, `os.File` implements `Reader`, so you don't even need to open a buffered reader for the file:
```
bytesWritten, err := io.Copy(ctx, file)
if err != nil {
    // handle errors
}
```